### ISSUE TITLE
Nothing rendered in window after MacPorts migration on Mojave

### ISSUE THREAD

**Issue by mdurian** [kivy/kivy#5987]
### Versions

* Python: 3.5.6
* OS: MacOS 10.14 (Mojave)
* Kivy: 1.10.1
* Kivy installation method: MacPorts (and pip)

### Description

I've been using Kivy for a while and it has been working for me. I updated my iMac to Mojave a week or two ago. I believe my kivy apps were still working after the update. Yesterday I migrated MacPorts to Mojave (MacPorts version 2.5.4) and now kivy isn't working for me any more. The application launches and a window appears with the correct title in the title bar, but nothing is rendered in the window itself. This isn't just my applications either. It happens using the Hello, World example shown here https://kivy.org/doc/stable/guide/basic.html.

I've uninstalled and reinstalled the ports a few times and it didn't help. I've also tried installing kivy using pip instead of MacPorts and it doesn't work there either. This is with kivy 1.10.1.

### Code and Logs

```python
import kivy
kivy.require('1.0.6') # replace with your current kivy version !

from kivy.app import App
from kivy.uix.label import Label


class MyApp(App):

    def build(self):
        return Label(text='Hello world')


if __name__ == '__main__':
    MyApp().run()
```

This is from the log file when I run the Hello, World script.

[INFO   ] Logger: Record log in /Users/durian/.kivy/logs/kivy_18-10-10_0.txt
[INFO   ] Kivy: v1.10.1
[INFO   ] Python: v3.5.6 (default, Oct  3 2018, 19:57:36)
[GCC 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.11.45.2)]
[INFO   ] Factory: 194 symbols loaded
[INFO   ] Image: Providers: img_tex, img_imageio, img_dds, img_sdl2, img_pil, im
g_gif (img_ffpyplayer ignored)
[INFO   ] Text: Provider: sdl2
[INFO   ] Window: Provider: sdl2
[INFO   ] GL: Using the "OpenGL ES 2" graphics system
[INFO   ] GL: Backend used <gl>
[INFO   ] GL: OpenGL version <b'2.1 NVIDIA-12.0.19 355.11.10.50.10.103'>
[INFO   ] GL: OpenGL vendor <b'NVIDIA Corporation'>
[INFO   ] GL: OpenGL renderer <b'NVIDIA GeForce GTX 775M OpenGL Engine'>
[INFO   ] GL: OpenGL parsed version: 2, 1
[INFO   ] GL: Shading version <b'1.20'>
[INFO   ] GL: Texture max size <16384>
[INFO   ] GL: Texture max units <16>
[INFO   ] Window: auto add sdl2 input provider
[INFO   ] Window: virtual keyboard not allowed, single mode, not docked
[INFO   ] Base: Start application main loop
[INFO   ] GL: NPOT texture support is available

**Comment by envil** [comment]
I've installed mojave freshly and it doesn't work either.

**Comment by huth314** [comment]
I did observe the same issue using Kivy 1.10.1 installed via homebrew on Mac OS 10.14 (Mojave). What I also found is that resizing the root widget causes the previously invisible widgets to become visible.

**Comment by dangerisom** [comment]
Same issue for me. Kivy installed via MacPorts. As with huth314, resizing or swiping does cause my invisible widgets to appear.

**Comment by alphazo-id** [comment]
Same thing here.
MacOS Mojave 10.14
Python 3.7.0, installed using homebrew
Kivy 1.10.1, installed using pip

At startup - black screen, after resize - widgets become visible.

**Comment by mdurian** [comment]
I can confirm that installing a newer sdl library fixed this problem. In my case, using MacPorts, I did a 'port selfupdate' followed by 'port upgrade outdated'. After that my kivy applications started appearing again.

**Comment by dangerisom** [comment]
I can confirm that mdurian's fix worked for me as well using MacPorts.

**Comment by huth314** [comment]
@mdurian 's approach (reinstall sdl2 with --HEAD option) did not work for me. I still have to resize to get to see the widgets.
MacOS Mojave 10.14.1
Python 3.7.0 installed via homebrew
Kivy 1.10.1 installed via pip3

**Comment by dangerisom** [comment]
To clarify my previous post...hope this helps. 
MacOS Mojave 10.14.1
Python 2.7.15 installed via MacPorts
Kivy 1.10.1 installed via MacPorts

**Comment by mgbowman** [comment]
@huth314 I have a a similar setup and managed to fix this!

macOS 10.14.1
Python 2.7.15 installed via pyenv (installed via Homebrew)
Kivy 1.10.1 installed via pip (within a pyenv-virtualenv)

I believe this issue is resolved in SDL 2.0.9 but there is an [upstream issue with the sdl2 formula for older versions of macOS](https://github.com/Homebrew/homebrew-core/pull/33781). We have to wait until this gets fixed before SDL 2.0.9 is available via `brew`. 

For now, you can use the `HEAD` version of SDL2 via `brew`. This is what worked for me...

```
$ brew uninstall sdl2 sdl2_image sdl2_ttf sdl2_mixer
$ brew install --HEAD sdl2
$ brew install sdl2_image sdl2_ttf sdl2_mixer
$ pip install --no-binary Kivy Kivy
```

Using the binary package from pip did not work so I had to add the `--no-binary Kivy` argument to force pip to build Kivy from source.

Hope this helps!

**Comment by tito** [comment]
Yep, just reinstall sdl and it's OK!
